### PR TITLE
Tests and updates for curriculum assignment Amplitude events

### PR DESCRIPTION
**Pull request by bethanyaconnor**
This PR is a follow up to https://github.com/code-dot-org/code-dot-org/pull/50033. It does a couple things:
1. Adds tests for the work in that PR
2. Adds courseVersionId to all instances of this event
3. Updates `EditSectionDialog` to use the shared constant

### CODE REVIEW

File: apps/src/templates/teacherDashboard/EditSectionForm.jsx [hunk 1]
```diff
@@ -186,15 +186,17 @@ class EditSectionForm extends Component {
         section.courseOfferingId !== initialCourseOfferingId) ||
         (section.unitId && section.unitId !== initialUnitId))
     ) {
-      analyticsReporter.sendEvent(CURRICULUM_ASSIGNED, {
+      analyticsReporter.sendEvent(EVENTS.CURRICULUM_ASSIGNED, {
```
Comment (hannahbergam) [inline apps/src/templates/teacherDashboard/EditSectionForm.jsx:189]: Thank you for updating this!

File: apps/test/unit/templates/teacherDashboard/teacherSectionsReduxTest.js [hunk 2]
```diff
@@ -112,8 +118,10 @@ const sections = [
     pairing_allowed: false,
     sharing_disabled: false,
     course_offering_id: 3,
+    courseOfferingId: 3,
     course_version_id: 5,
     unit_id: 7,
+    unitId: 7,
```
Comment (hannahbergam) [inline apps/test/unit/templates/teacherDashboard/teacherSectionsReduxTest.js]: Can you remind me what we need both versions of all these variables for (like unitId and unit_id)?

**Review by hannahbergam — APPROVED** [review]
Thanks for doing this follow up! One small question but LGTM

File: apps/test/unit/templates/teacherDashboard/teacherSectionsReduxTest.js (hunk 2 above)
Comment (bethanyaconnor) [reply apps/test/unit/templates/teacherDashboard/teacherSectionsReduxTest.js]: good catch! I thought we needed this but I'm pretty sure we don't. I'll remove

**Pull request merged**